coder-sdk/env.go

```diff
@@ -138,6 +161,49 @@ func (c Client) DialEnvironmentBuildLog(ctx context.Context, envID string) (*web
 	return c.dialWebsocket(ctx, "/api/environments/"+envID+"/watch-update")
 }
 
+// BuildLog defines a build log record for a Coder environment.
+type BuildLog struct {
+	ID            string `db:"id" json:"id"`
+	EnvironmentID string `db:"environment_id" json:"environment_id"`
+	// BuildID allows the frontend to separate the logs from the old build with the logs from the new.
+	BuildID string       `db:"build_id" json:"build_id"`
+	Time    time.Time    `db:"time" json:"time"`
+	Type    BuildLogType `db:"type" json:"type"`
+	Msg     string       `db:"msg" json:"msg"`
+}
+
+// BuildLogFollowMsg wraps the base BuildLog and adds a field for collecting
+// errors that may occur when follow or parsing.
+type BuildLogFollowMsg struct {
+	BuildLog
+	Err error
+}
+
+// FollowEnvironmentBuildLog trails the build log of a Coder environment.
+func (c Client) FollowEnvironmentBuildLog(ctx context.Context, envID string) (<-chan BuildLogFollowMsg, error) {
+	ch := make(chan BuildLogFollowMsg)
+	ws, err := c.DialEnvironmentBuildLog(ctx, envID)
```

Comment:
@f0ssel @coadler Any thoughts on this pattern? Seems better to me than having a `BuildLogWatcher` type with methods for iterating when channels are designed for that.